src/mod/applications/mod_http_cache/mod_http_cache.c

```diff
@@ -768,13 +771,22 @@ static char *url_cache_get(url_cache_t *cache, http_profile_t *profile, switch_c
 
 		/* download the file */
 		url_cache_unlock(cache, session);
+		clock_gettime(CLOCK_REALTIME, &before);
 		if (http_get(cache, profile, u, use_mime_ext, event, session) == SWITCH_STATUS_SUCCESS) {
+			unsigned int duration;
+			clock_gettime(CLOCK_REALTIME, &after);
+			duration = (after.tv_sec - before.tv_sec) * 1000 + lround((after.tv_nsec - before.tv_nsec) * 1.0e6);
+			prometheus_increment_download_duration(duration);
+			prometheus_increment_download_count();
+
 			/* Got the file, let the waiters know it is available */
 			url_cache_lock(cache, session);
 			u->status = CACHED_URL_AVAILABLE;
 			filename = switch_core_strdup(pool, u->filename);
 			cache->size += u->size;
 		} else {
+			prometheus_increment_download_fail_count();
```

Comment:
Take note that http_get handle a retry mechanism for download failures. Meaning that if http_get retries 3 times where the download fail count will only report it 1.